components/rusoto_util/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::env::{var, VarError};
+use std::io::{self, Error, ErrorKind};
+
+use rusoto_core::{
+    region::Region,
+    request::{HttpClient, HttpConfig},
+};
+use rusoto_credential::{
+    AutoRefreshingProvider, AwsCredentials, ChainProvider, CredentialsError, ProvideAwsCredentials,
+    StaticProvider, Variable,
+};
+use rusoto_sts::WebIdentityProvider;
+
+use async_trait::async_trait;
+
+const READ_BUF_SIZE: usize = 1024 * 1024 * 2;
+
+#[macro_export]
+macro_rules! new_client {
+    ($client: ty, $config: ident) => {{
+        let http_client = $crate::new_http_client()?;
+        new_client!($client, $config, http_client)
+    }};
+    ($client: ty, $config: ident, $dispatcher: ident) => {{
+        let region = $crate::get_region($config.region.clone(), $config.endpoint.clone())?;
+        let cred_provider = $crate::CredentialsProvider::new(
+            $config.access_key.clone(),
+            $config.secret_access_key.clone(),
+        )?;
+        <$client>::new_with($dispatcher, cred_provider, region)
+    }};
+}
+
+pub fn new_http_client() -> io::Result<HttpClient> {
+    let mut http_config = HttpConfig::new();
+    // This can greatly improve performance dealing with payloads greater
+    // than 100MB. See https://github.com/rusoto/rusoto/pull/1227
+    // for more information.
+    http_config.read_buf_size(READ_BUF_SIZE);
+    HttpClient::new_with_config(http_config).map_err(|e| {
+        Error::new(
+            ErrorKind::Other,
+            format!("create aws http client error: {}", e),
+        )
+    })
+}
+
+pub fn get_region(region: String, endpoint: String) -> io::Result<Region> {
+    if !endpoint.is_empty() {
+        Ok(Region::Custom {
+            name: region,
+            endpoint,
+        })
+    } else if !region.is_empty() {
+        region.parse::<Region>().map_err(|e| {
+            Error::new(
+                ErrorKind::InvalidInput,
+                format!("invalid aws region format {}: {}", region, e),
+            )
+        })
+    } else {
+        Ok(Region::default())
+    }
+}
+
+pub enum CredentialsProvider {
+    Default(Box<AutoRefreshingProvider<DefaultCredentialsProvider>>),
+    Static(StaticProvider),
+}
+
+impl CredentialsProvider {
+    pub fn new(access_key: String, secret_access_key: String) -> io::Result<CredentialsProvider> {
```

Comment:
```suggestion
    pub fn new(access_key: &str, secret_access_key: &str) -> io::Result<CredentialsProvider> {
```

And use `.to_string()` / `.to_owned()` for lines 77-78.